packages/commerce-sdk-react/src/hooks/ShopperLogin/helper.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ShopperLoginTypes} from 'commerce-sdk-isomorphic'
+import {ActionResponse, Argument} from '../types'
+import {useAsyncCallback} from '../useAsync'
+import useAuth from '../useAuth'
+import Auth from '../../auth'
+
+export enum ShopperLoginHelpers {
+    LoginGuestUser = 'loginGuestUser',
+    LoginRegisteredUserB2C = 'loginRegisteredUserB2C',
+    Logout = 'logout'
+}
+
+/**
+ * A hook for Public Client Shopper Login OAuth helpers.
+ * The hook calls the SLAS helpers imported from commerce-sdk-isomorphic.
+ * For more, see https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/#public-client-shopper-login-helpers
+ *
+ * Avaliable helpers:
+ * - loginRegisteredUserB2C
+ * - loginGuestUser
+ * - logout
+ */
+export function useShopperLoginHelper(
+    action: ShopperLoginHelpers.LoginRegisteredUserB2C
+): ActionResponse<Parameters<Auth['loginRegisteredUserB2C']>, ShopperLoginTypes.TokenResponse>
+export function useShopperLoginHelper(
+    action: ShopperLoginHelpers.LoginGuestUser
+): ActionResponse<never, ShopperLoginTypes.TokenResponse>
+export function useShopperLoginHelper(
+    action: ShopperLoginHelpers.Logout
+): ActionResponse<never, ShopperLoginTypes.TokenResponse>
+export function useShopperLoginHelper(
+    action: ShopperLoginHelpers
+): ActionResponse<
+    Parameters<Auth['loginRegisteredUserB2C']> | never,
+    ShopperLoginTypes.TokenResponse
+> {
```

Comment:
```suggestion
export function useShopperLoginHelper<Action extends `${ShopperLoginHelpers}`>(
    action: Action
): ActionResponse<Parameters<Auth[Action]>, ShopperLoginTypes.TokenResponse> {
```
TypeScript enums have weird rules. Using `(action: ShopperLoginHelpers)` means you _must_ use the `ShopperLoginHelpers` enum. This means, for example, that `useShopperLoginHelper('logout')` will report an error, even though we know it works at runtime. Using ``<Action extends `${ShopperLoginHelpers}`>`` gets around this weird legacy behavior and enables passing strings directly.

Also, it's shorter!

Reply:
the suggestion looks pretty neat, but i hit a few roadblocks for this... gonna look into this later

Reply:
✅ fixed!